zapcore/console_encoder.go

```diff
@@ -79,8 +79,8 @@ func (c consoleEncoder) EncodeEntry(ent Entry, fields []Field) (*buffer.Buffer,
 	if c.LevelKey != "" && c.EncodeLevel != nil {
 		c.EncodeLevel(ent.Level, arr)
 	}
-	if ent.LoggerName != "" && c.NameKey != "" {
-		arr.AppendString(ent.LoggerName)
+	if ent.LoggerName != "" && c.NameKey != "" && c.EncodeLoggerName != nil {
+		c.EncodeLoggerName(ent.LoggerName, arr)
```

Comment:
Unfortunately, this isn't backward-compatible. If `c.EncodeLoggerName` is nil, we should use `FullLoggerNameEncoder`.